Add documentation for Result (#413)
icerpc/icerpc-docs: content/slice/encoding/constructed-types.md

Author: It's just alphabetical order.

---
title: Constructed types
description: Learn how to encode constructed types with Slice.
---

{% slice1 %}
Slice provides two built-in generic types: `Dictionary<Key, Value>` and `Sequence<T>`. You need to specify the type
argument(s) of such a generic type in order to get an actual type - a constructed type. This page describes the encoding
of such constructed types.
{% /slice1 %}

{% slice2 %}

> Slice provides three built-in generic types: `Dictionary<Key, Value>`, `Result<Success, Failure>` and `Sequence<T>`. You

Author: Fixed

`Key` represents the dictionary's key type and `Value` represents the dictionary's value type.

{% slice2 %}

## Result

A `Result<Success, Failure>` is encoded exactly like a compact enum with two enumerators (`Success` and `Failure`),

> where each enumerator has a single field of the Success resp. Failure type.

Reviewer: I found the end of this sentence unclear, is this what its trying to say?

```suggestion
where each enumerator has a corresponding single field, the success response, or the failure type.
```

Reviewer 2: Honestly, I find it perfectly readable, and better than the suggestion... But you don't see this `resp.` syntax used much nowadays. I think just writing it out would be fine.
```suggestion
where each enumerator has a single field of the respective (Success or Failure) type.
```

Reviewer: I didn't even know resp. was shorthand for respective so I think this definitely needs to be changed. Im okay with either @InsertCreativityHere or my suggestion!

Author: Fixed

> where each enumerator has a single field of the Success resp. Failure type.

Reviewer 2: Honestly, I find it perfectly readable, and better than the suggestion... But you don't see this `resp.` syntax used much nowadays. I think just writing it out would be fine.
```suggestion
where each enumerator has a single field of the respective (Success or Failure) type.
```
